ocaml/lambda/printlambda.ml

```diff
@@ -220,39 +230,54 @@ let layout ppf layout_ = layout' true ppf layout_
 let return_kind ppf (mode, kind) =
   let smode = alloc_mode_if_local mode in
   match kind with
-  | Pvalue { raw_kind = Pgenval; _ } when is_heap_mode mode -> ()
-  | Pvalue { raw_kind = Pgenval; _ } -> fprintf ppf ": %s@ " smode
-  | Pvalue { raw_kind = Pintval; _ } -> fprintf ppf ": int@ "
-  | Pvalue { raw_kind = Pboxedfloatval bf; _ } ->
-     fprintf ppf ": %s%s@ " smode (boxed_float_name bf)
-  | Pvalue { raw_kind = Parrayval elt_kind; _ } ->
-     fprintf ppf ": %s%sarray@ " smode (array_kind elt_kind)
-  | Pvalue { raw_kind = Pboxedintval bi; _ } -> fprintf ppf ": %s%s@ " smode (boxed_integer_name bi)
-  | Pvalue { raw_kind = (Pboxedvectorval (Pvec128 v)); _ } ->
-    fprintf ppf ": %s%s@ " smode (vec128_name v)
-  | Pvalue { raw_kind = (Pvariant { consts; non_consts; }); _ } ->
-    variant_kind (value_kind value_kind_non_null') ppf ~consts ~non_consts
+  | Pvalue { raw_kind; nullable } -> begin
+    let or_null_suffix =
+      match nullable with
+      | Non_nullable -> ""
+      | Nullable -> " or_null"
+    in
+    match raw_kind with
+    | Pgenval when is_heap_mode mode -> ()
+    | Pgenval -> fprintf ppf ": %s@ " smode
+    | Pintval -> fprintf ppf ": int@ "
```

Comment:
In all these cases we don't print the `or_null` information, is this intentional?

Reply:
The `int` case is a mistake; I'll fix it.